### DISCUSSION TITLE
How to run tests in the same order every time?

### DISCUSSION THREAD

**Discussion by ronny-lark** [vitest-dev/vitest#4448] [Q&A]
My tests run in a somewhat random order, very loosely close to their name but not always.
The order is random on every run, which is particularly frustrating for me because we have a test that causes a CI pipeline to hang indefinitely, but it doesn't happen anywhere else.
I cannot pin down the exact test since there's no indication it started or ended or otherwise. I tried Printing the test name in `beforeEach` but it only printed tests that ran successfully, meaning the issue must be in the `describe` scope which I can't pin down.
If the tests ran in the same order every time everywhere, I would be able to figure out which test is messing up by its order.
I tried running with `vitest --sequence.seed 2023` but that literally did nothing.
How can I make the tests run in the same exact order every time?

**Comment by sheremet-va** [comment]
Sequence.seed will work only if sequencer is random

Default sequencer runs tests that failed first

**Reply by ronny-lark** [reply]
So how do I both set the seed and set the sequencer to random form cli?

**Reply by sheremet-va** [reply]
https://vitest.dev/config/#sequence-shuffle
